cmd/simulator/txs/agent.go

```diff
@@ -75,11 +92,71 @@ func (a issueNAgent[T]) Execute(ctx context.Context) error {
 			}
 			txs = append(txs, tx)
 		}
+		issuedEnd := time.Now()
+		issuedDuration := issuedEnd.Sub(issuedStart)
+		log.Info("Issuance Batch Done", "batch", batchI, "time", issuedDuration.Seconds())
+		// Add the issuance batch time to the total issuedTime
+		issuedTime = issuedTime.Add(issuedDuration)
 
+		// Start confirmation batch
+		confirmedStart := time.Now()
 		for i, tx := range txs {
 			if err := a.worker.ConfirmTx(ctx, tx); err != nil {
 				return fmt.Errorf("failed to await transaction %d: %w", i, err)
 			}
+			confirmedCount++
+			if confirmedCount == totalTxs {
+				// Mark the final ending time
+				confirmedEnd := time.Now()
+				totalTime := time.Since(start).Seconds()
+				// Get the last confirmed batch time and add it to the total confirmedTime
+				confirmedDuration := confirmedEnd.Sub(confirmedStart)
+				log.Info("Confirmed Batch Done", "batch", batchI, "time", confirmedDuration.Seconds())
+				confirmedTime = confirmedTime.Add(confirmedDuration)
+
+				// Get the final duration by comparing it to the zero time
+				issuedFinalDuration := issuedTime.Sub(time.Time{})
+				confirmedFinalDuration := confirmedTime.Sub(time.Time{})
+
+				log.Info("Execution complete", "totalTxs", totalTxs, "totalTime", totalTime, "TPS", float64(totalTxs)/totalTime,
+					"issuanceTime", issuedFinalDuration.Seconds(), "confirmedTime", confirmedFinalDuration.Seconds())
```

Comment:
nit: I don't think we need to call `.Seconds()`, we can log the duration directly

Reply:
.If I remove the Seconds() we will get different units. For example, issuanceTime (ms) and confirmationTime(s). I want to keep them standard.